app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt

```diff
@@ -56,10 +56,11 @@ class GetMessagesForConversationUseCase @Inject constructor(
             pagingConfig = pagingConfig
         ).map { pagingData ->
             pagingData.flatMap { messageItem ->
-                observeMemberDetailsByIds(messageMapper.memberIdList(listOf(messageItem)))
-                    .mapLatest {
-                        messageMapper.toUIMessages(it, listOf(messageItem))
-                    }.first()
+                val uiMessages = mutableListOf<UIMessage>()
+                observeMemberDetailsByIds(messageMapper.memberIdList(listOf(messageItem))).mapLatest { usersList ->
+                    messageMapper.toUIMessage(usersList, messageItem)?.let { uiMessages.add(it) }
+                }
+                uiMessages
```

Comment:
This looks a bit weird...

We create a mutable list and return it empty, but we map values from a flow and add them to the mutable list.

However, this flow is not even being collected, so it shouldn't even be working 🤔 

As in the [docs](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/):
> Intermediate operators on the flow such as [map](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/map.html), [filter](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/filter.html), [take](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/take.html), [zip](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/zip.html), etc [...]. **Intermediate operations do not execute any code in the flow and are not suspending functions themselves. They only set up a chain of operations for future execution and quickly return.** This is known as a cold flow property.


Perhaps this works because the flow in question is _hot_. So it's kind of unexpected, really.

I'd prefer the `.first()` how it was over this. As at least it will keep working if we change the flow implementation inside Kalium.